packages/ap-storage/alliance_platform/storage/fields/async_file.py

```diff
@@ -0,0 +1,839 @@
+    :class:`~alliance_platform.storage.views.DownloadRedirectView`. This view will check the user has permission to download the
+    file by checking the ``perm_download`` permission. If not provided this defaults to the value returned by
+    :meth:`django_site_core.auth.resolve_perm_name` with the action "detail" (ie. if they have permission to view the record
+    they can download the file). This view will then redirect to the URL provided by
+    :meth:`~alliance_platform.storage.storage.AsyncUploadStorage.generate_download_url`.
+
+    .. note:: You must you this with a storage class that implements :class:`~alliance_platform.storage.storage.AsyncUploadStorage`.
+        Either set DEFAULT_FILE_STORAGE to a class (eg. :class:`~alliance_platform.storage.s3.S3AsyncUploadStorage`) or pass an
+        instance in the ``storage`` kwarg.
+
+    For use with a Presto form use :class:`~alliance_platform.storage.drf.serializer.AsyncFileField` or :class:`~alliance_platform.storage.drf.serializer.AsyncImageField`
+    on your serializer. This is handled by default when extending :class:`xenopus_frog_app.base.XenopusFrogAppModelSerializer`.
+    Codegen will create this as a ``AsyncFileField`` or ``AsyncImageField`` on the frontend Presto model.
+    ``getWidgetForField`` will map these fields to the ``UploadWidget``.
+
+    See :class:`~alliance_platform.storage.fields.async_file.AsyncFileField` and :class:`~alliance_platform.storage.fields.async_file.AsyncFileField`
+    for Field classes that use this.
+
+    .. warning:: User input should be assigned using the :class:`~alliance_platform.storage.fields.async_file.AsyncFileInputData` class
+        which has validation to disallow things like moving the file by changing the key that passed from the frontend. This
+        is handled for you when using :class:`~alliance_platform.storage.fields.async_file.AsyncFileField`,
+        :class:`~alliance_platform.storage.fields.async_file.AsyncImageField`, :class:`~alliance_platform.storage.drf.serializer.AsyncFileField`
+        or :class:`~alliance_platform.storage.drf.serializer.AsyncImageField`
+
+    If a file cannot be moved for any reason (eg. any exception is raised) then the error will be saved in the
+    ``error`` field on the :class:`~alliance_platform.storage.models.AsyncTempFile` record and the temporary key will be retained as
+    the field value. The error will also be logged to the ``alliance_platform.storage`` logger. This cannot be resolved automatically
+    and so requires manual intervention to cleanup.
+
+    File size can be restricted by passing in ``max_size`` (in MB)
+
+    The file extension/file type can be restricted by passing in ``file_restrictions`` for example:
+
+    .. code-block:: python
+
+        ["image/png"]
+        ["image/*"]
+        [".doc", "docx"]
+        ["application/vnd.openxmlformats-officedocument.wordprocessingml.document"]
```

Comment:
Would it make sense to move some or most of this overview to its own docs section? Whether moved or not, it should be linked prominently on the index to make it easier to find.

Reply:
For now it's exposed in the API section, but I'm planning on restructuring the docs before release so will see about moving it then